src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,34 @@
+import { popupSaveButton } from "../utils/constants";
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupSaveButton = this._popupElement.querySelector(
+      ".popup__save-button"
+    );
+    this._popupSaveButtonText = this._popupSaveButton.textContent;
+  }
+
+  close() {
+    this._popupSaveButton.removeEventListener("click", this._handleFormSubmit);
```

Comment:
Needs correction: there is no need to remove the listener, since you don't dd it when opening the popup